unison-src/transcripts/reflog.output.md

```diff
@@ -59,16 +59,16 @@ y = 2
   most recent, along with the command that got us there. Try:
   
     `fork 2 .old`             
-    `fork #o7ncp4f3j1 .old`   to make an old namespace
+    `fork #t80e5h32dh .old`   to make an old namespace
```

Comment:
Not sure whether all these hashes are expected to change, I suppose it's possible that re-ordering operations could result in a new state, but I'll have to look into it to see if it's equivalent, or if I broke or fixed something 🤔

Reply:
Let's walk through the algorithm together before merging — we do want a multiple history steps on the sub-branches to be merged into one, with a squash step after all the actions have been run but before we finish and return to the input loop.

Reply:
I could believe that it's appropriate for some of these hashes to change though, we should dig into it though and see what the actual difference between the two histories is.

Reply:
Ah I see, so I should make sure I've run all changes (even from the parent branch) before committing the change to a new causal.

At a glance I think the old code might be making a mistake there too, since the action over the "root" branch might still make changes to the children after they've "stepped" from their own changes, but I should be able to fix up both of those (although it might still result in new hashes if I can get it making exactly one history step haha)

Reply:
Yeah the existing code definitely has a bug that caused #1430 / #1495.